pkg/client/client_test.go

```diff
@@ -1 +1,89 @@
 package client
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials/provider"
+	sdkerrors "github.com/aliyun/alibaba-cloud-sdk-go/sdk/errors"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+)
+
+var (
+	testCreads = `
+	[default]
+	type = access_key
+	access_key_id = YourAccessKeyId
+	access_key_secret = YourAccessKeySecret
+	`
+)
+
+func getCredentials() (auth.Credential, error) {
+	creds := []byte(testCreads)
+	f, err := ioutil.TempFile("", "alibaba-creds-*")
+	if err != nil {
+		return nil, err
+	}
+
+	defer os.Remove(f.Name())
+	defer f.Close()
+
+	_, err = f.Write(creds)
+	if err != nil {
+		return nil, err
+	}
+
+	os.Setenv(provider.ENVCredentialFile, f.Name())
+	defer os.Unsetenv(provider.ENVCredentialFile)
+	// use Alibaba provider initialization
+	p := provider.NewProfileProvider("default")
+	// return a valid auth credential
+	return p.Resolve()
+}
+
+func Test_EcsEndPoint(t *testing.T) {
```

Comment:
I'm not really sure what you're trying to do with this test, as far as I can tell it isn't testing any code within the cluster api provider?

Reply:
My main purpose was to verify that the endpoint was working as expected. So i add this test case. We can remove it if we have to

Reply:
So does this test run through each of the endpoints and attempt to connect to them? Checks that the endpoints haven't changed and that the SDK is up to date? 

IMO that doesn't seem like the kind of test we would need for this project, it seems like it is likely to break in the future through no fault of our own and that would then cause issues for us as we have to fix the test

Reply:
I got it. so i will remove this test.